forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -2662,6 +2662,27 @@ else if (sq[0].contains("Remembered")) {
             return doXMath(manaCost.size(), expr, c, ctb);
         }
 
+        //Count$DifferentManaValue<zone> <restriction>
+        if (l[0].contains("DifferentManaValue")) {
```

Comment:
I think it might be worth it to wire this up with `handlePaid` instead if there's another _$_ so you can just use
`Count$ValidGraveyard Card.YouOwn+nonLand$DifferentCMC`

That could provide some nice cleanup potential in other places too :)

Reply:
Will this be done as a part of this MR?

Reply:
I just made it an issue; I'd rather focus elsewhere for now